lib/spack/spack/cmd/bootstrap.py

```diff
@@ -32,6 +34,16 @@ def _add_scope_option(parser):
 def setup_parser(subparser):
     sp = subparser.add_subparsers(dest='subcommand')
 
+    status = sp.add_parser('status', help='get the status of Spack')
+    status.add_argument(
+        '--optional', action='store_true', default=False,
+        help='show the status of rarely used optional dependencies'
+    )
+    status.add_argument(
+        '--dev', action='store_true', default=False,
+        help='show the status of dependencies needed to develop Spack'
+    )
```

Comment:
Looks like the PR could use at least a couple of unit tests (e.g., exercising these options).